src/Language/PureScript/TypeChecker/TypeSearch.hs

```diff
@@ -85,6 +110,9 @@ typeSearch
   -- ^ The typechecker state
   -> P.Type
   -- ^ The type we are looking for
-  -> Map (P.Qualified P.Ident) P.Type
+  -> ([(P.Qualified P.Ident, P.Type)], Maybe [(Label, Type)])
```

Comment:
Is there a good reason to include `_.y` there? I think we could exclude it by unifying the type of the label with the result type.